### ISSUE TITLE
"Diff with Vim" Context Menu Option Missing in Windows 11 after Installing Vim 9.1

### ISSUE THREAD

**Issue by thomasdn** [vim/vim#15673]
### Steps to reproduce

After installing Vim 9.1 on Windows 11, the "Diff with Vim" option is no longer available in the right-click context menu in File Explorer. This issue occurs in both the default Windows 11 context menu and the legacy context menu (accessed by holding Shift + right-click).

## Steps to reproduce:

1. Install Vim 9.1 on Windows 11.
2. Right-click on any two files in File Explorer.
3. Check both the default context menu and the legacy context menu (Shift + right-click) for the "Diff with Vim" option.
4. Observe that the "Diff with Vim" option is missing.

Additional Notes:
Is there a workaround or a method to manually restore the "Diff with Vim" option in the context menu? Any guidance on how to get the functionality back would be helpful. This feature was present in earlier versions of Vim and is now missing after the upgrade to Vim 9.1.

### Expected behaviour

Context menu item "Diff with vim" expected to be there. (same as Open with Vim which seems to be gone as well)

The "Diff with Vim" option should appear in the right-click context menu in Windows Explorer, allowing users to easily compare two files using vimdiff.

Actual Behavior:
The context menu option is missing after upgrading to Vim 9.1. This makes it difficult to use vimdiff for comparing files without manually running commands in the terminal.

### Version of Vim

VIM - Vi IMproved 9.1 (2024 Jan 02, compiled Jan  2 2024 23:51:27)

### Environment

Vim Version: 9.1
OS: Windows 11 (latest updates applied)

VIM - Vi IMproved 9.1 (2024 Jan 02, compiled Jan  2 2024 23:51:27)
MS-Windows 64-bit GUI version with OLE support
Compiled by appveyor@APPVEYOR-VM

**Comment by chrisbra** [comment]
I cannot reproduce it. I still have it with the [Vim installer](https://github.com/vim/vim-win32-installer/releases) Can you try that one please?

**Comment by thomasdn** [comment]
> I cannot reproduce it. I still have it with the [Vim installer](https://github.com/vim/vim-win32-installer/releases) Can you try that one please?

Is this the same as this one which is linked to on www.vim.org?
https://github.com/vim/vim-win32-installer/releases/download/v9.1.0/gvim_9.1.0_x64_signed.exe

If so, I have just tested using this one and I can reproduce it.

**Comment by thomasdn** [comment]
> I don't remember any changes specifically to the installer, but I'd ask you to test with the latest version please

Right. I just tried with the latest gvim_9.1.0727_x64.exe

I chose "full installation" during install. I made sure to verify that the "Add 'open with vim...' context menus" option was checked during install.

Still, the context menu items are not present.

**Comment by thomasdn** [comment]
I just tried this on a completely different machine running Windows 11 as well and using the newest installer: gvim_9.1.0727_x64.exe

Same result: context menus are not present.

I don't have anymore Win11 machines at my disposal to test this on. But so far, this issue has been present on 2 out of 2 systems I tested.

I am not sure if there is a way to create a free Win11 VM like there was with Win10. If there is, I may be able to test on more systems...

**Comment by chrisbra** [comment]
What version did you try? What about 32 vs 64 bit? What about running as 
admin user?

Thanks
Chris


Am 2024-09-12 14:34, schrieb thomasdn:

**Comment by thomasdn** [comment]
> What version did you try? What about 32 vs 64 bit? What about running as admin user?

64 bit, non-admin user.

**Comment by k-takata** [comment]
Currently, the installer can install the context menu items only when it is executed as an admin user.

**Comment by thomasdn** [comment]
> Currently, the installer can install the context menu items only when it is executed as an admin user.

Interesting. I will see if I can try using admin user.

I would propose one of the following changes to fix this issue then:

1. Make the installer prompt for elevated privileges during install if this feature is checked.
2. Show a warning to the user that the context menus will be missing
3. If possible, figure out a way to implement the context menus without admin privileges. Not sure if possible. But I could imagine admin privileges being required if this is a system-wide change. However, if the installation is for a single user, perhaps it is possible to add context menus only for the user running the installer?

**Comment by thomasdn** [comment]
> Currently, the installer can install the context menu items only when it is executed as an admin user.

Now I just had the opportunity to do a reinstall. I used the gvim_9.1.0733_x64.exe installer and ran it as Administrator. (Started CMD as Administrator, then ran the installer from cmd)

Even in this case, I do not see the context menu entries. Neither in regular context menu nor in legacy context menu.

I did not completely uninstall my current Vim installation first. I just reinstalled on top of the old one to upgrade in-place.

**Comment by k-takata** [comment]
Hmm, I've just confirmed that gvim_9.1.0733_x64.exe installs the context menu items in the legacy context menu on Windows 11.
The menu items are only installed in the legacy context menu, so you must hold the shift key when you click the right button.

**Comment by thomasdn** [comment]
> Hmm, I've just confirmed that gvim_9.1.0733_x64.exe installs the context menu items in the legacy context menu on Windows 11. The menu items are only installed in the legacy context menu, so you must hold the shift key when you click the right button.

Right. I have tested this, and I do not get the context menus in the legacy menu. 
So something seems to not work correctly. 
I tested this by reinstalling Vim as Administrator. So perhaps there is some issue relating to cleanup/reset when doing reinstall? Perhaps it is seeing that some entry is still there and skips installing it again?

**Comment by EldarYuzbashev** [comment]
I had the same problem. Didn't realize that the presence of the Vim context menu item depends on the "cloud" status of the file. If the status icon is the "cloud" the Vim menu is not present. Otherwise, it's there. "Diff with Vim" and other Vim context menu items are also present.
Vim 9.1.1891 x64, Windows 11 24H2 (OS Build 26100.6899)
<img width="448" height="478" alt="Image" src="https://github.com/user-attachments/assets/aa756eba-50d6-42a5-947f-b6df70ed8882" />
<img width="462" height="443" alt="Image" src="https://github.com/user-attachments/assets/36eed3e2-4481-4474-a318-1d87d4694b33" />
